branding_company/model/stock_picking.py

```diff
@@ -26,19 +25,31 @@ class StockPicking(models.Model):
     """Extend stock.picking with branding."""
     _inherit = 'stock.picking'
 
-    def _prepare_invoice(
-            self, cr, uid, picking, partner, inv_type, journal_id,
-            context=None):
+    def _compute_branding(self):
+        """Branding for a delivery order is taken from sales order.
+
+        We use the procurement group to find all relates sale orders.
+        The first sale order found will determine the branding.
+        """
+        sale_model = self.env['sale.order']
+        sale_orders = sale_model.search([
+            ('procurement_group_id', '=', self.group_id)])
+        return (
+            sale_orders and
+            sale_orders[0].branding_company and
+            sale_orders[0].branding_company.id or False
```

Comment:
@StefanRijnhart Relying on False being returned when accessing an undefined attribute seems to me that we will base our code on an accidental and strange implementation detail of the new API. 
@hbrunn I would be amazed if this really does not error out on an empty recordset. Anyhow, it would still seem to me to be relying on an implementation detail of recordsets. Or is there something in the public API documentation that guarantees this will work?